Formula/p/pan.rb

```diff
@@ -0,0 +1,43 @@
+class Pan < Formula
+  desc "Usenet newsreader that's good at both text and binaries"
+  homepage "https://pan.rebelbase.com"
+  url "https://gitlab.gnome.org/GNOME/pan/-/archive/v0.154/pan-v0.154.tar.bz2"
+  sha256 "440317954df7217689100df3dfb68865770f5aed1b8ed2b45432d771bb80a8c9"
+  license "GPL-2.0-only"
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "itstool" => :build
+  depends_on "pkg-config" => :build
+  depends_on "adwaita-icon-theme"
+  depends_on "cairo"
+  depends_on "enchant"
+  depends_on "gdk-pixbuf"
+  depends_on "gettext"
+  depends_on "glib"
+  depends_on "gmime"
+  depends_on "gnutls"
+  depends_on "gtk+3"
+  depends_on "gtkspell3"
+  depends_on "harfbuzz"
+  depends_on "pango"
+
+  def install
+    # use brew name for gtk3 version of tool update-icon-cache
+    inreplace  "pan/icons/Makefile.am", "gtk-update-icon-cache", "gtk3-update-icon-cache"
+
+    ENV.append "LDFLAGS", "-liconv" if OS.mac?
+    ENV.append "CXXFLAGS", "-std=c++11"
+
+    system "NOCONFIGURE=1 ./autogen.sh"
+    system "./configure", *std_configure_args,
+                          "--disable-silent-rules",
+                          "--with-gnutls"
+    system "make"
+    system "make", "install"
+  end
+
+  test do
+    assert_match shell_output("#{bin}/pan --version"), "Pan" + " " + version.to_s
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
To be honest, that's why I hesitated a while to bring this one to you... I thought the test would be a hiccup.

This is a newsreader application. To exercise it, you need to configure a news server, download the newsgroup list, and then interact with it.

But I can't think of a way to make it do those those things as part of a test run. It does have some non-gui functionality, but as far as I can see, that only works after it's been configured with a news server, etc. 

I looked at some older legacy Formula for newsreaders to see what they did -- like tin.rb for example, and some of them got away with just a verson test, so I thought I'd see how you liked it.

It's OK if the lack of a good test makes this inappropriate for homebrew. I understand how that would be.

Reply:
That sounds like it would be a good fit for this suggestion:

> If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.

Reply:
I have asked the pros:

https://lists.nongnu.org/archive/html/pan-users/2023-10/msg00006.html